agent/agent.go

```diff
@@ -444,7 +444,7 @@ func (a *Agent) handleSessionMessage(ctx context.Context, message *api.SessionMe
 		if !same {
 			a.keys = message.NetworkBootstrapKeys
 			if err := a.config.Executor.SetNetworkBootstrapKeys(a.keys); err != nil {
-				panic(fmt.Errorf("configuring network key failed"))
+				return errors.Wrap(err, "configuring network key failed")
```

Comment:
Unrelated to this fix, but this logic seems a little odd:

```
	for _, key := range message.NetworkBootstrapKeys {
		same := false
		for _, agentKey := range a.keys {
			if agentKey.LamportTime == key.LamportTime {
				same = true
			}
		}
		if !same {
			a.keys = message.NetworkBootstrapKeys
			if err := a.config.Executor.SetNetworkBootstrapKeys(a.keys); err != nil {
				return errors.Wrap(err, "configuring network key failed")
			}
		}
	}
```

It looks like we are looping over each bootstrap key, and for each key that is different, we set the agent keys to the bootstrap keys - so if multiple keys are different, we'd set the same set of keys multiple times.

This seems like an idempotent action, so it's probably not an issue, but it does give the executor more changes to fail.

Happy to change this in a different PR, though.

Reply:
Is setting the network bootstrap keys an expensive operation?  Can we just set it every time?  cc @abhi

Reply:
From slack conversation - we're not sure whether it's expensive.